### PR TITLE
Update dependency versions

### PR DESCRIPTION
**Pull request by arcondello**
Build and support dimod~=0.11.0
Increase NumPy lower bound to 1.19.1
Drop support for Python 3.6

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def build_extensions(self):
         dimod.get_include(),
         ],
     install_requires=[
-        'numpy>=1.17.3,<2.0.0',  # keep synced with circle-ci, pyproject.toml
-        'dimod>=0.10.0,<0.11.0'
+        'numpy>=1.19.1,<2.0.0',  # keep synced with circle-ci, pyproject.toml
+        'dimod>=0.11.0rc0,<0.12.0'
```
Comment (randomir) [inline setup.py:64]: Why not `<0.13.0` here as well?

Comment (randomir) [inline setup.py:64]: IIRC, `>=...rc` will trigger pip's `--pre` (for other packages as well).

Comment (arcondello) [reply setup.py:64]: We need binary compatibility for this one.

Comment (arcondello) [reply setup.py:64]: Hmm, not sure there's much to be done about that. I think it's relatively safe in this case since it's just NumPy?

Comment (randomir) [reply setup.py:64]: Maybe all other packages if installed as part of Ocean? Not sure. But we can test and update if needed.

**Review by randomir — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (arcondello) [reply setup.py:64]: One obvious thing is just release this package as an `.rc` and and then when we release a non-`.rc` version of dimod we also update this one.

Comment (arcondello) [reply setup.py:64]: And only the non-`.rc` will make it into a proper sdk release.

**Pull request merged**